### DISCUSSION TITLE
Trying to use lightning deposit address xpub for BTC Pay Server

### DISCUSSION THREAD

**Discussion by jaxkeiser** [lightningnetwork/lnd#6996] [Q&A]
I want to get the public key of my lightning wallet's deposit address so that I can enter it on my BTC Pay Server and receive payments to my lightning wallet's on-chain deposit address.

I used the command "lncli wallet accounts list" and it displayed a list of xpubs. I had to use one of the last ones on the list due to a weird issue with the terminal not letting me scroll up that far.

I entered it into BTC Pay Server and sent a small payment. The payment is fully confirmed on the blockchain, but the funds never arrived to my lighting wallet on-chain.

What am I doing wrong?

**Comment by guggero** [comment]
You need to use the correct xpub depending on the address type you use.
 - `"address_type": "HYBRID_NESTED_WITNESS_PUBKEY_HASH",` (`"derivation_path": "m/49'/0'/0'",`) is for NP2WKH (legacy segwit) addresses.
 - `"address_type": "WITNESS_PUBKEY_HASH",` (`"derivation_path": "m/84'/0'/0'",`) is for P2WKH (native segwit) addresses.
 - `"address_type": "TAPROOT_PUBKEY",` (`"derivation_path": "m/86'/0'/0'",`) is for P2TR (native Taproot) addresses.
 - Everything else (`"derivation_path": "m/1017'/1'/X'",`) is `lnd` internal stuff.

Seems like you caught one of the last ones, which won't ever carry funds directly.
Try piping the output to another program (e.g. `lncli wallet accounts list | more`) to paginate the output. Or query the address type you want directly with `lncli wallet accounts list --address_type p2wkh`.
And by the way, `--help` is always your friend when using the command line (e.g. `lncli wallet accounts list --help`).

**Reply by jaxkeiser** [reply]
Thanks bud <3

**Reply by jaxkeiser** [reply]
By the way, I should never have to change the xpub, correct? It will continue working?

**Reply by guggero** [reply]
No, you don't need to change the xpub. But lnd will only detect the addresses it already derived itself. So you'll need to call `lncli newaddress <addr_type>` as many times as you plan on receiving to different addresses. And you need to do it before you receive, otherwise lnd won't see the funds until you do a full manual rescan.
